hschain-PoW/HSChain/PoW/Node.hs

```diff
@@ -89,17 +91,13 @@ runNode pathsToConfig miningNode genesisBlock step inventHeaderTxs inventBlock s
                       , nConnectedPeers = 3
                       }
   let net = newNetworkTcp cfgPort
-  db <- inMemoryDB @_ @_ @b
+  db <- inMemoryDB genesisBlock
```

Comment:
I didn't touch runNode beyond necessary since it's not clear whether it will survive in the long term. I feels it tries to do too many things at once and it would be beneficial to split it into smaller functions but it's not clear what those should be